csvy/writers.py

```diff
@@ -310,6 +311,43 @@ def write_polars(
     return False
 
 
+@register_writer
+def write_dict(
+    filename: Path | str,
+    data: Any,
```

Comment:
How about:

```suggestion
    data: Mapping[str, Any],
```

Reply:
The problem is that, then, I should type annotate all the other readers AND the registry with exactly what they ingest, which is not possible to do for arrays and dataframes unless the relevant optional dependencies are included. So, I rather annotate with `Any` to skip the static checking there and leave proper type checking at runtime.

Reply:
Ohh I see. Nvm then.